gridstatus/ercot.py

```diff
@@ -432,6 +501,253 @@ def get_interconnection_queue(self, verbose=False):
 
         return queue
 
+    def get_lmp(
+        self,
+        date,
+        end=None,
+        market: str = None,
+        locations: list = None,
+        location_type: str = None,
+        verbose=False,
+    ):
+        """Get LMP data for ERCOT
+
+        Supported Markets: REAL_TIME_15_MIN, DAY_AHEAD_HOURLY
+
+        Supported Location Types: "zone", "hub", "node"
+        """
+        if not (date == "latest" or utils.is_today(date)):
+            raise NotSupported(f"date={date} is not supported for LMP")
+
+        if locations is None:
+            locations = "ALL"
+
+        if location_type is None:
+            location_type = LOCATION_TYPE_ZONE
+
+        assert market is not None, "market must be specified"
+        market = Markets(market)
+
+        unsupported = False
+        if market == Markets.REAL_TIME_15_MIN:
+            if date == "latest":
+                return self._get_lmp_rtm15_latest(
+                    locations,
+                    location_type,
+                    verbose,
+                )
+            elif utils.is_today(date):
+                return self._get_lmp_rtm15_today(
+                    locations,
+                    location_type,
+                    verbose,
+                )
+            else:
+                unsupported = True
+        elif market == Markets.DAY_AHEAD_HOURLY:
+            if date == "latest":
+                return self._get_lmp_dam_latest(locations, location_type, verbose)
+            elif utils.is_today(date):
+                return self._get_lmp_dam_today(locations, location_type, verbose)
+            else:
+                unsupported = True
+        else:
+            unsupported = True
+
```

Comment:
I think some form of the following logic is in each of the `_get_spp` methods. can we just include it in this function to avoid repetition and increase maintainability?

```python
df = self._filter_by_locations(df, "SettlementPointName", locations)

df = df.rename(
    columns={
        "SettlementPointPrice": "LMP",
        "SettlementPointName": "Location",
    },
)

df = df[
    [
        "Location",
        "Time",
        "Market",
        "Location Type",
        "LMP",
    ]
]

df = df.reset_index(drop=True)```